pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -770,6 +772,39 @@ type SNSConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+// TelegramConfig configures notifications via Telegram.
+// See https://prometheus.io/docs/alerting/latest/configuration/#telegram_config
+type TelegramConfig struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The Telegram API URL i.e. https://api.telegram.org.
+	// If not specified, default API URL will be used.
+	// +optional
+	APIURL string `json:"apiURL,omitempty"`
+	// Telegram bot token
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +kubebuilder:validation:Required
+	BotToken *v1.SecretKeySelector `json:"botToken,omitempty"`
+	// The Telegram chat ID.
+	// +kubebuilder:validation:Required
+	ChatID int64 `json:"chatID,omitempty"`
+	// Message template
+	// +optional
+	Message string `json:"message,omitempty"`
+	// Disable telegram notifications
+	// +optional
+	DisableNotifications *bool `json:"disableNotifications,omitempty"`
+	// Parse mode for telegram message
+	//+kubebuilder:validation:Enum="";MarkdownV2;Markdown;HTML
```

Comment:
nit: since this is marked as optional, I don't believe we need to include the empty string in the enum

Reply:
Ye! I was having doubts with the enum, I looked for examples in the project and I found: 
https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/thanos_types.go#L145
So I decided to leave it with the empty value as it is a valid value documented: https://prometheus.io/docs/alerting/latest/configuration/#telegram_config

Reply:
Still, if you think it could be better to remove it... let me know and I'll do it

Reply:
I did a quick test and indeed the Kube API makes a distinction between 
* "the field is not present in the submitted request" :heavy_check_mark:
* "the field is present and the value is empty" :no_entry_sign:
* "the field is present and the value is in the enums list" :heavy_check_mark: 

From an API perspective, it's probably better to remove "" from the allowed values but we can tackle in a follow-up PR and globally.